docs/examples/low_level/ingestion.ipynb

```diff
@@ -0,0 +1,545 @@
+   "id": "fac468f5-870c-4486-b576-2aa6d9eaf322",
+   "metadata": {},
+   "source": [
+    "### [Optional] 4. Extract Metadata from each Node\n",
+    "\n",
+    "We extract metadata from each Node using our Metadata extractors.\n",
+    "\n",
+    "This will add more metadata to each Node."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 29,
+   "id": "9c369188-9cc9-4550-924e-b29d212ad057",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from llama_index.node_parser.extractors import (\n",
+    "    MetadataExtractor,\n",
+    "    QuestionsAnsweredExtractor,\n",
+    "    TitleExtractor,\n",
+    ")\n",
+    "from llama_index.llms import OpenAI\n",
+    "\n",
+    "llm = OpenAI(model=\"gpt-3.5-turbo\")\n",
+    "\n",
+    "metadata_extractor = MetadataExtractor(\n",
+    "    extractors=[\n",
+    "        TitleExtractor(nodes=5, llm=llm),\n",
+    "        QuestionsAnsweredExtractor(questions=3, llm=llm),\n",
+    "    ],\n",
+    "    in_place=False,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "f5501ffc-9bbb-4b48-9181-4e4e371e8f41",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "nodes = metadata_extractor.process_nodes(nodes)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9d52522c-ffee-49d1-8651-658d70248053",
+   "metadata": {},
+   "source": [
+    "### 5. Generate Embeddings for each Node\n",
+    "\n",
+    "Generate document embeddings for each Node using our OpenAI embedding model (`text-embedding-ada-002`).\n",
+    "\n",
+    "Store these on the `embedding` property on each Node."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 30,
+   "id": "6e071e36-e609-4a0c-a478-e8cfbe751cff",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from llama_index.embeddings import OpenAIEmbedding\n",
+    "\n",
+    "embed_model = OpenAIEmbedding()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 33,
+   "id": "2047ca46-729f-4c5a-a8d7-3bc860604333",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "for node in nodes:\n",
+    "    node_embedding = embed_model.get_text_embedding(\n",
+    "        node.get_content(metadata_mode=\"all\")\n",
+    "    )\n",
+    "    node.embedding = node_embedding"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "11f78014-dcca-43f5-95cb-9cfb5140b30e",
+   "metadata": {},
+   "source": [
+    "### 6. Load Nodes into a Vector Store\n",
+    "\n",
+    "We now insert these nodes into our `PineconeVectorStore`.\n",
+    "\n",
+    "**NOTE**: We skip the VectorStoreIndex abstraction, which is a higher-level abstraction that handles ingestion as well. We use `VectorStoreIndex` in the next section to fast-trak retrieval/querying."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "fe34fbe4-3396-402e-8599-4b42013c3016",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "vector_store.add(nodes)"
```

Comment:
For some reason, I run into a problem at this line
`AttributeError: 'TextNode' object has no attribute 'id'`
Seems like vector_store takes in nodes of type `NodeWithEmbedding` which has id property, 
but the nodes passed in here are TextNode that doesn't have `id` property